src/foam/dao/DAOProperty.js

```diff
@@ -27,7 +27,18 @@ foam.CLASS({
   properties: [
     {
       name: 'view',
-      value: {class: 'foam.comics.InlineBrowserView'},
+      value: { class: 'foam.comics.InlineBrowserView' },
+    },
+    {
+      name: 'readView',
```

Comment:
I'm not too keen on having a readView on property because it means that I'll often have to set both view and readView at the same time. I would rather have the view decide. For convenience, we could have a ReadWriteView like:
view: { class: 'foam.u2.ReadWriteView', readView: ...,  writeView }
But if I have a custom view for colours, URLs, telephone numbers, etc. I shouldn't have to know to set two views, one for edit and one for view mode.

Reply:
> I'll often have to set both view and readView at the same time.

You only have to set it if you're not satisfied with the default values of those properties for your property type. Which is the same as it already is with just `view`. It's quite rare that you'd want to set both to the same view anyway.

> But if I have a custom view for colours, URLs, telephone numbers, etc. I shouldn't have to know to set two views, one for edit and one for view mode.

But with what you're suggesting, you'd still have to know to set `view` to `ReadWriteView` instead of your custom view directly. Either way you need to be aware of the distinction between the read view and the write view, so moving it doesn't buy you anything.

I don't particularly like the idea of having just `view` because almost every property's view will become `ReadWriteView`, and if you want to override it you'll need to copy the one of the two that you don't want to override.

For example, if we did what you're suggesting, the `view` of `FObjectProperty` would become

```
{
  class: 'foam.u2.view.ReadWriteView',
  readView: {
    class: 'foam.u2.view.CitationView'
  },
  writeView: {
    class: 'foam.u2.DetailView'
  }
}
```

and now if I want to override the read view for any FObjectProperty I need to set it to

```
{
  class: 'foam.u2.view.ReadWriteView',
  readView: {
    class: 'my.custom.thing.CustomView'
  },
  // Manually copy and paste this from the default because I don't want to change this
  writeView: {
    class: 'foam.u2.DetailView'
  }
}
```

rather than simply setting the `readView` directly to the custom view and leaving the write view as its default.

Reply:
@kgrgreer Troy's argument is sound.

Reply:
@kgrgreer and I talked in person and decided to stick with just `view` so that the "API" of views doesn't change. Each property has a single view, and it's up to that view to decide what to show based on the mode of the view. Therefore we created `ModeAltView`, which shows different views based on the mode. A new view was created for each property type (or at least most of them) that extends `ModeAltView` and sets the `readView` and `writeView` to whatever is appropriate for that type. New views were created where there were no suitable ones already available.